src/loader/mod.rs

```diff
@@ -293,10 +293,11 @@ impl KernelLoader for Elf {
                 .read_exact_from(mem_offset, kernel_image, phdr.p_filesz as usize)
                 .map_err(|_| Error::ReadKernelImage)?;
 
-            loader_result.kernel_end = mem_offset
+            let kernel_end = mem_offset
```

Comment:
Is there a specific type of segment will have such issue? Refer to [elf](https://refspecs.linuxbase.org/elf/elf.pdf), it says "Loadable segment entries in the program header table appear in ascending order, sorted on the p_vaddr member" under PT_LOAD.

Reply:
There's no concrete example, just for safety:)